docs: Update README with more information about the framework (#70)
wpengine/faustjs: README.md

# Headless WordPress Framework

## Introduction

> WPEngine's Headless WordPress Framework provides a set of tools to make building front-end applications with WordPress as the headless CMS a pleasant experience for both developers and publishers. This framework consists of a WordPress plugin, a set of npm packages, and guides to get you started building headless WordPress sites in Next.js.

Reviewer: 'WPEngine' → 'WP Engine' (to follow brand guidelines).

```suggestion
WP Engine's Headless WordPress Framework provides a set of tools to make building front-end applications with WordPress as the headless CMS a pleasant experience for both developers and publishers. This framework consists of a WordPress plugin, a set of npm packages, and guides to get you started building headless WordPress sites in Next.js.
```

_🚧 **Note:** This project is in the early stages of development_

## Quick Start

Eager to try out the Headless Framework? Here's how you can get started:

1. Create a WordPress site if you haven't already. We recommend using [Local](https://localwp.com/)!
2. Download, upload, and activate the `wpe-headless` plugin. [(Plugin Download)](https://wp-product-info.wpesvc.net/v1/plugins/wpe-headless?download)
3. Install [WP GraphQL](https://wordpress.org/plugins/wp-graphql/) on the WordPress site if it's not already installed

> 4. Download our [Getting Started example project](https://github.com/wpengine/headless-framework/tree/canary/examples/getting-started) and navigate to it locally

Reviewer: `getting-started` doesn't exist yet.

Should we leave this URL as https://github.com/wpengine/headless-framework/tree/canary/examples/preview until `getting-started` exists? @claygriffiths mentioned on Friday that we planned to rename `preview`.

Author: Ah yeah, I forgot we talked about that. That's a much better experience than manually donloading.

Author: Also, yeah I think we should rename `preview` to `getting-started`. For this PR though, yeah you're right I'll revert back to `preview` and we can update to `getting-started` once we make that change

> 6. `cp .env.local.sample .env.local`
> 7. Populate `NEXT_PUBLIC_WORDPRESS_URL` and `WPE_HEADLESS_SECRET` accordingly in `.env.local`
> 8. `npm install && npm run dev`

Reviewer: Perhaps make it more clear what `NEXT_PUBLIC_WORDPRESS_URL` should contain, and how to find the `WPE_HEADLESS_SECRET`:

```suggestion
6. `cp .env.local.sample .env.local`
7. Populate `NEXT_PUBLIC_WORDPRESS_URL` in `.env.local` with the full URL to your WordPress site, including the `http://` or `https://` prefix.
8. Populate the `WPE_HEADLESS_SECRET` in `.env.local` with the secret key found at Settings → Headless in your WordPress admin area.
9. `npm install && npm run dev`
```

Author: Good callouts, thanks!

## Framework Features

### Plugin Features

- **[Headless post previewing](./docs/previews/README.md)**
  - OAuth token authentication creation
  - Preview and draft link rewrites in WP Admin to redirect to the front-end
- **Smart content redirects**
  - Automatically redirects content from the WP site to the front-end site to minimize site visitors’ confusion and avoid SEO penalties for duplicate content
  - Redirects hyperlinks inserted into posts’ content to the front-end site
- **Removal of WP themes**
  - Prevents confusion around the active theme
- **Ability to define custom menus in a GUI**
- **Additional data exposed through WPGraphQL**
  - Block stylesheets

> ### npm Package Features

Author: Need to flesh the npm features out more for sure

Reviewer: I added some suggested descriptions for `HeadlessProvider` and `TemplateLoader` below.

Author: Thank you!

- Auth handler for Express/Next that exchanges a code for an access token
  - The access token can be used to make authenticated calls to WordPress via WPGraphQL or REST

> - `HeadlessProvider`

Reviewer: ```suggestion
- A `HeadlessProvider` component to ease communication with WordPress via [Apollo](https://www.apollographql.com/) and [WPGraphQL](https://www.wpgraphql.com/).
```

> - `TemplateProvider`

Reviewer: ```suggestion
- A `TemplateLoader` component to load page templates based on the current URL path and page type.
```

# Headless WordPress Framework

## Introduction

WP Engine's Headless WordPress Framework provides a set of tools to make building front-end applications with WordPress as the headless CMS a pleasant experience for both developers and publishers. This framework consists of a WordPress plugin, a set of npm packages, and guides to get you started building headless WordPress sites in [Next.js](https://nextjs.org/).

_🚧 **Note:** This project is in the early stages of development_

## Quick Start

Eager to try out the Headless Framework? Here's how you can get started:

1. Create a WordPress site if you haven't already. We recommend using [Local](https://localwp.com/)!
2. Download, upload, and activate the `wpe-headless` plugin. [(Plugin Download)](https://wp-product-info.wpesvc.net/v1/plugins/wpe-headless?download)

> 3. Install [WP GraphQL](https://wordpress.org/plugins/wp-graphql/) on the WordPress site if it's not already installed\

Reviewer 2: ```suggestion
3. Install [WP GraphQL](https://wordpress.org/plugins/wp-graphql/) on the WordPress site if it's not already installed
```

Author: Ope, good catch
